src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishASingleFileApp.cs

```diff
@@ -352,5 +353,80 @@ public void It_rewrites_the_apphost_for_non_single_file_publish()
 
             appHostSize.Should().BeLessThan(singleFileSize);
         }
+
+        //  Core MSBuild only due to https://github.com/dotnet/sdk/issues/4244
+        [CoreMSBuildOnlyFact]
+        public void It_leaves_host_components_unbundled_when_necessary()
+        {
+            // In.net 5, Single-file bundles are processed in the framework.
+            // Therefore, in self-contained builds, hostpolicy and hostfxr DLLs cannot themselves be in the bundle.
+            // This check is temporary until until statically linked singlefilehost is supported: 
+            // * https://github.com/dotnet/runtime/issues/32823
+            // * https://github.com/dotnet/sdk/issues/11567
+
+            var testProject = new TestProject()
+            {
+                Name = "SingleFileTest",
+                TargetFrameworks = "netcoreapp5.0",
+                IsSdkProject = true,
+                IsExe = true
+            };
+
+            var testAsset = _testAssetsManager.CreateTestProject(testProject);
+            var publishCommand = new PublishCommand(Log, Path.Combine(testAsset.TestRoot, testProject.Name));
+
+            publishCommand.Execute(PublishSingleFile, RuntimeIdentifier)
+                .Should()
+                .Pass();
+
+            string hostfxr = RuntimeInformation.RuntimeIdentifier.StartsWith("win") ? "hostfxr.dll" :
+                             RuntimeInformation.RuntimeIdentifier.StartsWith("osx") ? "libhostfxr.dylib" : "libhostfxr.so";
+
+            string hostpolicy = RuntimeInformation.RuntimeIdentifier.StartsWith("win") ? "hostpolicy.dll" :
+                                RuntimeInformation.RuntimeIdentifier.StartsWith("osx") ? "libhostpolicy.dylib" : "libhostpolicy.so";
+
+            string[] expectedFiles = { $"{testProject.Name}{Constants.ExeSuffix}", $"{testProject.Name}.pdb", hostfxr, hostpolicy };
+
+            GetPublishDirectory(publishCommand, "netcoreapp5.0")
+                .Should()
+                .OnlyHaveFiles(expectedFiles);
+        }
+
+        //  Core MSBuild only due to https://github.com/dotnet/sdk/issues/4244
+        [CoreMSBuildOnlyTheory]
```

Comment:
are all these combinations necessary too?

Reply:
Yes, for `It_runs_single_file_apps` I think we should have all combinations.
This guards against incompatibility between SDK and hostmodel that creates single-file apps that don't run correctly. I'm planning to add this test to [integration tests](https://github.com/dotnet/installer/blob/master/test/SdkTests/TestConfig.xml) in the installer branch, as recommended by Daniel.